### PR TITLE
Remove assign_id.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: pil-analyzer/src/condenser.rs [hunk 1]
```diff
@@ -91,19 +89,16 @@ pub fn condense<T: FieldElement>(
...
-    public_declarations
-        .values_mut()
-        .for_each(|public_decl| condenser.assign_id(&mut public_decl.polynomial));
+    for decl in public_declarations.values_mut() {
+        let symbol = &definitions
+            .get(&decl.polynomial.name)
+            .unwrap_or_else(|| panic!("Symbol {} not found.", decl.polynomial))
+            .0;
+        let reference = &mut decl.polynomial;
+        // TODO this is the only point we still allign poly_id,
```
Comment (leonardoalt) [inline pil-analyzer/src/condenser.rs:98]:
```suggestion
        // TODO this is the only point we still align poly_id,
```

Comment (leonardoalt) [reply pil-analyzer/src/condenser.rs:98]: in this PR?

Comment (chriseth) [reply pil-analyzer/src/condenser.rs:98]: This side-tracked me from a side-track, so probably not.

**Review by leonardoalt — APPROVED** [review]
please squash

**Pull request merged**